### PR TITLE
[PLAT-1104] Fix WB hook tests

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

We have these tests for these WB hooks that we don't use yet and they have gone un-run for awhile. In that time we've had several major changes to the site particularly Files-on-anything that have caused these tests to fail. This PR brings the tests back on track and addresses some tangential problems were discovered in the process.

## Changes

- add a migration to re-instate the file name uniqueness constraint that was silently broken by FOA
- removes non-functional `ShowIfVersion` field which had no effect on the node field (because it's a required write-only field) and replaces it with a conditional requirement in the serializer.
- Changes urls to use the current version of the API.
- Updates old `node` payload keys to `target`

## QA Notes

These tests should run on Travis and pass.

## Documentation

🐞 fix no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-1104

### CODE REVIEW

File: api/wb/serializers.py [hunk 1]
```diff
@@ -3,19 +3,22 @@
...
 from website.files import exceptions as file_exceptions
-from api.base.serializers import IDField, ShowIfVersion
+from api.base.serializers import IDField
 
 
 class DestinationSerializer(ser.Serializer):
     parent = ser.CharField(write_only=True)
     target = ser.CharField(write_only=True)
     name = ser.CharField(write_only=True, allow_blank=True, allow_null=True)
-    node = ShowIfVersion(
```
Comment (Johnetordoff) [inline api/wb/serializers.py]: As I noted in the PR description `ShowIfVersion` doesn't actually effect write only fields. I tried mightily to make a RequireIfVersion` field, but since in the field model we are working outside the request context this was difficult to do.

File: api/wb/serializers.py [hunk 2]
```diff
@@ -3,19 +3,22 @@
...
-        ser.CharField(write_only=True),
-        min_version='2.0', max_version='2.7',
-    )
+    node = ser.CharField(write_only=True)
 
+    def bind(self, *args, **kwargs):
+        super(DestinationSerializer, self).bind(*args, **kwargs)
+        version = self.parent.context['view'].request.version
+        self.fields['node'].required = not utils.is_deprecated(version, min_version='2.0', max_version='2.7')
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply api/wb/serializers.py]: This is excellent!

[the remark this answers is not in the document]
Comment (erinspace) [reply api/wb/serializers.py]: All you @Johnetordoff, I just moved it to a slightly different place! :D

File: osf/migrations/0142_reinstate_path_unique_constraint.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
```
Comment (sloria) [inline osf/migrations/0142_reinstate_path_unique_constraint.py]: Drive-by: This migration seems unrelated to this PR. Can you separate this into a different PR?

Comment (Johnetordoff) [reply osf/migrations/0142_reinstate_path_unique_constraint.py]: The tests won't pass without the migration, so if we split them up you'll have to run the migration before this gets merged.

Comment (sloria) [reply osf/migrations/0142_reinstate_path_unique_constraint.py]: Oh ok, I didn't realize this migration was related to the tests. OK, this is fine here then.

File: api/wb/serializers.py [hunk 4]
```diff
@@ -4,18 +4,13 @@
...
 
 class DestinationSerializer(ser.Serializer):
     parent = ser.CharField(write_only=True)
     target = ser.CharField(write_only=True)
     name = ser.CharField(write_only=True, allow_blank=True, allow_null=True)
-    node = ShowIfVersion(
-        ser.CharField(write_only=True),
-        min_version='2.0', max_version='2.7',
-    )
-
```
Comment (pattisdr) [inline api/wb/serializers.py]: Note on the `node` field for reviewers.

I recommended removing versioning of obsolete node field on DestinationSerializer. This was for a private endpoint, that was written only to be called by WaterButler through a signed endpoint. It is not even in use yet, because we've had other priorities. I would assume no one else is using it either. Regardless, I don't think we have a responsibility to version these private endpoints.

File: osf/migrations/0142_reinstate_path_unique_constraint.py (hunk 3 above)
Comment (pattisdr) [reply osf/migrations/0142_reinstate_path_unique_constraint.py]: I'm not convinced we need this migration with this PR - tests pass here https://github.com/CenterForOpenScience/osf.io/pull/8936.  We could merge that, and fix the broken data separately.
